### DISCUSSION TITLE
what layers does yggdrasil cover?

### DISCUSSION THREAD

**Discussion by boogerlad** [yggdrasil-network/yggdrasil-go#890] [Q&A]
![image](https://user-images.githubusercontent.com/1903750/151955075-05600716-9263-42ca-a197-09034825f39c.png)
From the GNUnet slides, it tries to replace the entire stack. It'll make more sense with the [presentation from 3:58 to 6:13 and 6:34 to 7:54](https://youtu.be/qZYI-3Q1INI?t=238)

Does yggdrasil just match circled layers?

**Comment by neilalexander** [comment]
Yggdrasil is an overlay network that operates on top of TCP, but provides IP-layer networking. It's not very easy to translate directly onto your diagram.
